### PR TITLE
:bug: Fixing issues of duplicate key mapping in serverless yml

### PR DESCRIPTION
**Pull request by SiniWilson**
### Requirements

This PR will help to resolve the issues of duplicate key mapping in serverless.yml file while integrating kinesis, sqs, and dynamodb events

### Description of the Change

Modified the serverless.yml files 
Modified the build-pack-api
Modified the build-pack-lambda

### Benefits

Successful build for event subscribed services and not subscribed services.

### CODE REVIEW

**Comment by devsatishm** [comment]
Can you update the description for the PR? What changes are we proposing with this PR?

File: builds/jenkins-build-pack-api/Jenkinsfile [hunk 1]
```diff
@@ -531,6 +531,9 @@ def writeServerlessFile(config, environment_logical_id){
 	sh "sed -i -- 's/\${file(deployment-env.yml):providerTimeout}/${config['providerTimeout']}/g' serverless.yml"
 	sh "sed -i -- 's/\${file(deployment-env.yml):securityGroupIds}/${config['securityGroupIds']}/g' serverless.yml"
 	sh "sed -i -- 's/\${file(deployment-env.yml):subnetIds}/${config['subnetIds'] }/g' serverless.yml"
+  sh "sed -i -- 's|\${file(deployment-env.yml):eventScheduleRate}|${config['eventScheduleRate']}|g' serverless.yml"
```
Comment (devsatishm) [inline builds/jenkins-build-pack-api/Jenkinsfile]: Why are these changes required in api build pack? Shouldn't we deal with these in lambda build pack instead?

File: builds/serverless-config-pack/serverless-java.yml [hunk 2]
```diff
@@ -87,28 +87,30 @@ functions:
...
     #Start:isDynamoDbEnabled
     customEventRole: ${file(policyFile.yml):eventSubscribed}
     #End:isDynamoDbEnabled
-    #Start:isSqsEnabled
+    #Start:isSQSEventEnabled
     customEventRole: ${file(policyFile.yml):sqsRole}
-    #End:isSqsEnabled
-    #Start:isSqsEnabled
+    #End:isSQSEventEnabled
+    #Start:isSQSEventEnabled
```
Comment (devsatishm) [inline builds/serverless-config-pack/serverless-java.yml]: Why is #Start:isSQSEventEnabled added again?

File: builds/jenkins-build-pack-api/Jenkinsfile (hunk 1 above)
Comment (devsatishm) [reply builds/jenkins-build-pack-api/Jenkinsfile]: If these are absolutely required, I would suggest removing the section instead of replacing with empty values (like in the case above).

File: builds/serverless-config-pack/serverless-java.yml (hunk 2 above)
Comment (SiniWilson) [reply builds/serverless-config-pack/serverless-java.yml]: The comment section added here is to remove the section when it is not using. The comment previously added was   #Start:isSqsEnabled   #End:isSqsEnabled But in lambda , used #Start:isSQSEventEnabled   #End:isSQSEventEnabled  to remove.  In order to fix that, I had renamed the comment.

File: builds/jenkins-build-pack-api/Jenkinsfile (hunk 1 above)
Comment (SiniWilson) [reply builds/jenkins-build-pack-api/Jenkinsfile]: Done. Removed the event scheduled as it is not required in API

File: builds/jenkins-build-pack-api/Jenkinsfile [hunk 3]
```diff
@@ -531,14 +531,15 @@ def writeServerlessFile(config, environment_logical_id){
 	sh "sed -i -- 's/\${file(deployment-env.yml):providerTimeout}/${config['providerTimeout']}/g' serverless.yml"
 	sh "sed -i -- 's/\${file(deployment-env.yml):securityGroupIds}/${config['securityGroupIds']}/g' serverless.yml"
 	sh "sed -i -- 's/\${file(deployment-env.yml):subnetIds}/${config['subnetIds'] }/g' serverless.yml"
+  sh "sed -i -- '/#Start:resources/,/#End:resources/d' ./serverless.yml"
+  sh "sed -i -- '/#Start:isScheduleEnabled/,/#End:isScheduleEnabled/d' ./serverless.yml"
```
Comment (devsatishm) [inline builds/jenkins-build-pack-api/Jenkinsfile]: How about other sections like `isDynamoDbEnabled`, `isSQSEventEnabled` etc. (actually everything under `eventsDisabled`)? Those are not required in API build workflows. Why are we only removing `isScheduleEnabled` section?

File: builds/serverless-config-pack/serverless-java.yml (hunk 2 above)
Comment (devsatishm) [reply builds/serverless-config-pack/serverless-java.yml]: I'm looking at code fragment below - 
`#End:isSQSEventEnabled`   
`#Start:isSQSEventEnabled`

Will the below work - ?

    `#Start:isSQSEventEnabled`
    `customEventRole: ${file(policyFile.yml):sqsRole}`
    `#Start:isSqsResourceNotExist`
    `{event_source_sqs}:`
      `Type: AWS::SQS::Queue`
      `Properties:`
        `QueueName: {event_source_sqs}`
        `VisibilityTimeout: 165`
    `#End:isSqsResourceNotExist`
    `#End:isSQSEventEnabled`

Comment (SiniWilson) [reply builds/serverless-config-pack/serverless-java.yml]: Oh ok understood your point. 
The SQS flow is , first  will add the sqs queue name in the resource part and will deploy, if the queue not exists, serverless will create the resource - by using the serverless yml file and will attach the lambda arn to the queue. Deployment will be success in this case.
2. If the queue already exists - Serverless will add the queue name in the resource part and will deploy, this time we will get an error while deploying saying resource already exists. So I will capture that error and remove the resource part from the serverless yml file and will redploy. So for removing the resource part only, I had added that comment section, since we need the custom role for both the case
